### ISSUE TITLE
[🐞]: Chipped (mod) variants of doors / trapdoors are uncraftable

### ISSUE THREAD

**Issue by unilock** [MehVahdJukaar/WoodGood#295]
### Before Continuing:

- [X] Always check you are using the latest version of the mods and its dependencies
- [X] Remove mod that enhances Minecraft: Optifine, Sodium, others. The issue still persists.
- [X] If you are unsure which mod is the culprit.
Disable all of your mods and enable them 1-2 mods each time to isolate the culprit
- [X] Confirm that there is no existing issue with a similar description submitted in the list of issues.

### Version

1.20.1

### Loader

Fabric

### EveryCompat Version

everycompat-1.20-2.6.10-fabric

### Moonlight Lib Version

moonlight-1.20-2.8.51-fabric

### Issue with mods

[Chipped v3.0.1](https://modrinth.com/mod/chipped/version/yXxHBIer)
and
[Promenade v4.1.1](https://modrinth.com/mod/promenade/version/v4.1.1)
[Regions Unexplored v0.5.3](https://modrinth.com/mod/regions-unexplored/version/0.5.3+1.20.1)
...?

### Issue Detail

Chipped (mod) variants of doors and trapdoors generated by Every Compat are uncraftable, where they would usually be crafted via Chipped's "Carpenter's Table".

I've noticed this issue with the doors and trapdoors from both Promenade and Regions Unexplored, though I suspect other mods would have the same issue.

Interestingly, each of Promenade's doors can be found in a tag including the door and every variant of it added by Every Compat, (e.g. `#everycomp:promenade_sakura_door` for sakura doors), but doors from Regions Unexplored have no such tags. (as seen in [EMI](https://modrinth.com/mod/emi))

### OPTIONAL: To Produce

1. Enter a world
2. Give yourself a "Carpenter's Table" (from Chipped) and i.e. a "Sakura Door" from Promenade
3. Open the Carpenter's Table and attempt to place the door in the input slot
4. The menu does not populate with possible door variants to craft, despite them existing via Every Compat

**Comment by Xelbayria** [comment]
thanks for providing the details, I'll run test on this

**Comment by Xelbayria** [comment]
The recipe system via **EC** is able to add **Doors**/**Trapdoors** to the **Carpentry's Table**. I've confirmed this on FORGE. but FABRIC's side is not working. 

I will investigate to find the cause. 

### EDIT: 
hmm, The system is not able to create the tags folder on Fabric.   

**Details:**
I remembered the chipped's system regarding the Carpentry's Table. It uses a JSON file which contains a list of doors/trapdoors and is placed in the tags folder for Chipped's system (like Carpentry's Table) to use as a reference. 

**Result from the investigation:**
I enabled Debug Resource (it created the resource pack folder) for me to check to see if the tags folder exists on FABRIC's side. There is no tags folder, unlike FORGE's side. 

Chipped's system has no files for reference.

**Comment by unilock** [comment]
Maybe I'm misunderstanding something, but it looks to me like every Chipped recipe is represented by a single tag in a corresponding `data/chipped/recipes/<workbench>.json` file.

In this case, the file would be `data/chipped/recipes/carpenters_table.json`, with the default being:  
https://github.com/terrarium-earth/Chipped/blob/1.20.x/common/src/main/generated/resources/data/chipped/recipes/carpenters_table.json

**Comment by Xelbayria** [comment]
Your link is the recipe for crafting `carpenters_table`. I can show you where are the files for doors, trapdoor, planks and others. 

[acacia_door.json](https://github.com/terrarium-earth/Chipped/blob/1.20.x/common/src/main/generated/resources/data/chipped/tags/blocks/acacia_door.json)
above is one JSON file for `acacia_door` and it has other doors with the same wood type. The file is representing a recipe for **acacia door** to be included in **Carpenter's Table**. 

The location of the files like above is via `chipped:tags/blocks/...` or `chipped:tags/items/...`.  That's what FABRIC's side is missing. I hope this helps you understand how the system of **Carpenter's Table** works.

**Comment by unilock** [comment]
> Your link is the recipe for crafting `carpenters_table`.

No, I don't think it is. This is what that file looks like formatted: https://gist.github.com/unilock/1d3ee956efb0a8fe4df0af749b48b1d6

If I'm understanding it correctly, the Carpenter's Table recipe serializer reads each tag in that file, and creates a new set of recipes per tag: each block in a given tag can be converted to any other block in that tag, and so on.

So, to add recipes to the Carpenter's Table, one would have to create a file `data/<namespace>/recipes/carpenters_table.json` with the following contents:
```json
{
    "type": "chipped:carpenters_table",
    "tags":
    [
        "namespace:tag_one",
        "namespace:tag_two",
        "namespace:tag_three",
        "..."
    ]
}
```
...with `namespace:tag_one`, `namespace:tag_two`, `namespace:tag_three`, etc. being tags that each contain a set of blocks that are meant to be converted between one another.

So in this case, for example:
```json
{
    "type": "chipped:carpenters_table",
    "tags":
    [
        "everycomp:promenade_sakura_door"
    ]
}
```
...would register Every Compat's Sakura Door (from Promenade) variants to Chipped's Carpenter's Table.

**Comment by unilock** [comment]
I can confirm that the above JSON for the Sakura Door works in-game :)

I just created a new datapack with the aforementioned JSON stored in `data/example/recipes/carpenters_table.json`.

<img width="668" alt="image" src="https://github.com/MehVahdJukaar/WoodGood/assets/11802133/9d3dd501-1fb7-4797-b92b-702a787a81da">

The datapack:  
[example.zip](https://github.com/MehVahdJukaar/WoodGood/files/13177419/example.zip)

**Comment by Xelbayria** [comment]
oh, I forgot about that. I had another look at the folder created by EC. **Recipes** folder containing the `carpenters_table.json` is there on the FORGE side, but not the FABRIC side. Not good. I don't know why the FABRIC side is not working.

That's good you created a solution for the problem. 

> >    Your link is the recipe for crafting carpenters_table.

> No, I don't think it is. 

You are correct, I was wrong. I forgot the purpose of that File is to list all of the blocks that will be used with `carpenters_table`
And the other files I am talking about are to keep them separated based on their wood types.

**Comment by gardenappl** [comment]
The `#everycomp:*_door` tags don't register for me on Forge, but I have quite a few mods installed, so it might take a while for me to debug.

**Comment by Xelbayria** [comment]
> The `#everycomp:*_door` tags don't register for me on Forge, but I have quite a few mods installed, so it might take a while for me to debug.

I am investigating this right now. I have a good idea why. I'm pinpointing the cause. 

It depends on the mods adding wood type.  It works fine with **Biomes O' Plenty**, but not with **Region Unexplored**. 

EDIT:
I finally understand why. The cause is a wood called "alpha" from **Region Unexplored**.  it doesn't have Trapdoor or door. Just logs, planks and slabs. that's it. 

The door or Trapdoor is missing. I'll apply a fix for this

**Comment by Xelbayria** [comment]
all fixed and working now. 

EDIT: 
confirmed on FABRIC, too.

**Comment by unilock** [comment]
Seems to be working here, thanks!

**Comment by CattsyTheKitty** [comment]
Hi, I'm experiencing this issue on Forge 1.19.2, Every Compat ver. 2.5.19. The Carpenter's Table from Chipped won't accept any modded doors or trapdoors. I checked on woodtypes added by Quark, Biomes O' Plenty, End's Phantasm, Natural Decor and Homespun. Is there any chance for a patch update for 1.19.2 as well?

**Comment by Xelbayria** [comment]
> Hi, I'm experiencing this issue on Forge 1.19.2, Every Compat ver. 2.5.19. The Carpenter's Table from Chipped won't accept any modded doors or trapdoors. I checked on woodtypes added by Quark, Biomes O' Plenty, End's Phantasm, Natural Decor and Homespun. Is there any chance for a patch update for 1.19.2 as well?

@CattsyTheKitty 
have you updated Moonlight Lib to the latest? & I'll check

**Comment by CattsyTheKitty** [comment]
> @CattsyTheKitty have you updated Moonlight Lib to the latest? & I'll check

I have Moonlight Lib ver. 2.3.6, which looks like the latest version for Forge 1.19.2

**Comment by Xelbayria** [comment]
I see. I've checked. It looks like Builder's Delight has been updated to add Trapdoor and door to _Carpenter's Table_. 

i'll add this in the next version. and also 1.20.1, too.

**Comment by CattsyTheKitty** [comment]
Thanks! I'll be looking out for a new version to drop and post feedback after testing.

**Comment by Xelbayria** [comment]
Ignore my previous message.   i was focused on **Builder's Delight**, not Chipped 🤦 

I took a look at the **chipped** via 1.19.2, it was fine. i'll check the wood mods to see if they have issue. BOP (BIomes O plenty) should be fine.  i'll check the others.

**Comment by Xelbayria** [comment]
Something caused the system to create a recipe for Carpenter's Table to fail and halt. This will take a while for me to find the cause and fix it. Thanks for reporting it.

**Comment by Xelbayria** [comment]
@CattsyTheKitty it's fixed and will be in the next version.

**Comment by CattsyTheKitty** [comment]
> Ignore my previous message. i was focused on **Builder's Delight**, not Chipped 🤦
> 
> I took a look at the **chipped** via 1.19.2, it was fine. i'll check the wood mods to see if they have issue. BOP (BIomes O plenty) should be fine. i'll check the others.

Haha np, my previous statement still stands 😊 Thank you for looking deep into this issue!

**Comment by Xelbayria** [comment]
@CattsyTheKitty 
Forgot to tell you. I thought the fixes from 1.20.1 to 1.19.2 were applied, but nope. It's fixed.
